aws/subscriber_test.go

```diff
@@ -185,3 +185,34 @@ func TestEnsureFilterPolicy_DifferentPolicy_SetsSubscriptionAttributes(t *testin
 		t.Error("expected SetSubscriptionAttributes to be called, but wasn't")
 	}
 }
+
+// TestDeleteSubscription verifies that a sqs queue can be deleted successfully
+func TestDeleteSubscription(t *testing.T) {
+	sqsMock := mockSQS{}
+	s, serr := newSubscriber(&mockSNS{}, &sqsMock, "topic", "subscriptionID")
+	if serr != nil {
+		t.Errorf("expected no error from newSubscriber, but got: %v", serr)
+	}
+
+	sqsMock.stubbedDeleteQueueError = errors.New("test error")
+	expected := s.DeleteSubscription()
```

Comment:
naming things: this should be `actual` or, better yet, `actualErr`

Reply:
Naming the error value anything other than `err` is a bit unconventional in the Go world, isn't it? I see named error values in a few places (not limited to this pull request). Not trying to be nit-picky, but it seems worth mentioning.

Reply:
that's a good callout in general. I think in this particular case, however, calling it `actualErr` is more meaningful because we're trying to test that calling a function returns a specific type of error. In that case, we'll have `expectedErr` and `actualErr`.